runners/prism/java/src/main/java/org/apache/beam/runners/prism/PrismLocator.java

```diff
@@ -54,34 +54,61 @@ class PrismLocator {
   private static final String PRISM_BIN_PATH = ".apache_beam/cache/prism/bin";
   private static final Set<PosixFilePermission> PERMS =
       PosixFilePermissions.fromString("rwxr-xr-x");
-  private static final String GITHUB_DOWNLOAD_PREFIX =
-      "https://github.com/apache/beam/releases/download";
-  private static final String GITHUB_TAG_PREFIX = "https://github.com/apache/beam/releases/tag";
+  private static final String GITHUB_COMMON_PREFIX = "https://github.com/apache/beam/releases/";
+  private static final String GITHUB_DOWNLOAD_PREFIX = GITHUB_COMMON_PREFIX + "download";
+  private static final String GITHUB_TAG_PREFIX = GITHUB_COMMON_PREFIX + "tag";
 
   private final PrismPipelineOptions options;
 
   PrismLocator(PrismPipelineOptions options) {
     this.options = options;
   }
 
+  String resolveLocation() {
```

Comment:
Perhaps we can call this `resolveFrom` or `resolveFromLocation` and keep the below `resolve` unchanged.